src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1463,19 +1435,58 @@ public boolean checkIfRequestForItemExistOrCreateAsynch(@NotNull final ItemStack
      * @param predicate the predicate to evaluate.
      * @return true if succesful.
      */
-    private boolean tryTransferFromPosToWorker(final BlockPos pos, @NotNull final Predicate<ItemStack> predicate)
+
+    private boolean tryTransferFromPosToWorker(final BlockPos pos, @NotNull final Tuple<Predicate<ItemStack>, Integer> predicate)
```

Comment:
make it two parameters isntead of the tuple

Reply:
The Itemlist created at  is a Tuple<Predicate<ItemStack>, Integer> because the needsCurrently var as the transfer var is feeded in the workers. and the Predicate<ItemStack> doesnt transfer the count.